ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -401,45 +402,37 @@ export default class ConfirmTransactionBase extends Component {
       </div>
     ) : null;
 
+    const renderLedgerLiveStep = (text, show = true) => {
+      return (
+        show && (
+          <Typography
+            boxProps={{ margin: 0 }}
+            color={COLORS.PRIMARY3}
+            fontWeight={FONT_WEIGHT.BOLD}
+            variant={TYPOGRAPHY.H7}
+          >
+            {text}
+          </Typography>
+        )
+      );
+    };
+
     const ledgerInstructionField = isLedgerAccount ? (
       <div>
         <div className="confirm-detail-row">
           <Dialog type="message">
             <div className="ledger-live-dialog">
-              <Typography
-                boxProps={{ margin: 0 }}
-                color={COLORS.PRIMARY3}
-                fontWeight={FONT_WEIGHT.BOLD}
-                variant={TYPOGRAPHY.H7}
-              >
-                {t('ledgerLiveDialogHeader')}
-              </Typography>
-              {!isFirefox && (
-                <Typography
-                  boxProps={{ margin: 0 }}
-                  color={COLORS.PRIMARY3}
-                  fontWeight={FONT_WEIGHT.BOLD}
-                  variant={TYPOGRAPHY.H7}
-                >
-                  {`- ${t('ledgerLiveDialogStepOne')}`}
-                </Typography>
+              {renderLedgerLiveStep(t('ledgerLiveDialogHeader'))}
+              {renderLedgerLiveStep(
+                `- ${t('ledgerLiveDialogStepOne')}`,
+                !isFirefox,
+              )}
+              {renderLedgerLiveStep(`- ${t('ledgerLiveDialogStepTwo')}`)}
+              {renderLedgerLiveStep(`- ${t('ledgerLiveDialogStepThree')}`)}
+              {renderLedgerLiveStep(
```

Comment:
Could you make it so that this is conditionally shown based on whether the current transactions has contract data or not?

Reply:
ah okay, all that needs to happen is we need to use `txData.txParams.data` instead of `methodData`. `methodData` is the results of calling `getKnownMethodData`, which returns a property from `state.metamask.knownMethodData`, which is not guaranteed to be truthy, even when sending to a contract. But `txData.txParams.data` will always be truthy when sending to a contract